### ISSUE TITLE
update tune docs and tests to use new validation set code

### ISSUE THREAD

**Issue by topepo** [tidymodels/tune#774]
There are places in the docs that mention `validation_split()` and also some tests

```
══ Warnings ══════════════════════════════════════
── Warning ('test-bayes.R:510:3'): missing performance values ──────────────────
`validation_split()` was deprecated in rsample 1.2.0.
i Please use `initial_validation_split()` instead.
Backtrace:
    ▆
 1. └─rsample::validation_split(ames, prop = 0.9) at test-bayes.R:510:3

── Warning ('test-fit_best.R:107:3'): fit_best() works with validation split: 2x 2-way splits ──
`validation_split()` was deprecated in rsample 1.2.0.
i Please use `initial_validation_split()` instead.
Backtrace:
    ▆
 1. └─rsample::validation_split(train_and_val) at test-fit_best.R:107:3
── Warning ('test-int_pctl.R:80:3'): percentile intervals - tuning ─────────────
`validation_split()` was deprecated in rsample 1.2.0.
i Please use `initial_validation_split()` instead.
Backtrace:
    ▆
 1. └─rsample::validation_split(two_class_dat) at test-int_pctl.R:80:3
```

**Comment by simonpcouch** [comment]
Duplicate of #731! The linked issue there notes what needs to happen in rsample before we can fix those warnings here.
